zerver/lib/streams.py

```diff
@@ -1114,6 +1122,37 @@ def bulk_can_remove_subscribers_from_streams(
     return True
 
 
+def bulk_can_unsubscribe_self_from_streams(
+    streams: list[Stream], user_profile: UserProfile
+) -> bool:
+    if user_profile.is_realm_admin:
+        return True
+
+    user_recursive_group_ids = set(
+        get_recursive_membership_groups(user_profile).values_list("id", flat=True)
+    )
+
+    permission_failure_streams: bool = True
```

Comment:
Since we do not need to store streams here in this variable, we can use some other name `cannot_administer_atleast_one_stream`.